src/lambdas/users-delete.js

```diff
@@ -0,0 +1,57 @@
+import ExceptionHandler from '@/exceptions/handler';
+import ResourceNotFoundException from '@/exceptions/resource-not-found';
+import User from '@/models/user';
+import {DataMapper} from '@aws/dynamodb-data-mapper';
+
+const DyanamoDB = require('aws-sdk/clients/dynamodb');
+const Lambda = require('aws-sdk/clients/lambda');
+
+/**
+ * @api {delete} /users/:id Delete a User
+ * @apiGroup Users
+ *
+ * @apiExample {curl} Example
+ * curl -X DELETE -H 'x-api-key: API_KEY_HERE' {{API_URL}}/users/bde526c8-9b59-45dc-9551-f8996d091fdc
+ *
+ * @apiSampleRequest off
+ *
+ * @apiUse Headers
+ *
+ * @apiParam {String} id User's unique identifier.
+ *
+ * @apiSuccessExample Success-Response
+ * HTTP/1.1 204 No Content
+ *
+ * @apiUse ResourceNotFoundException
+ * @apiUse Exception
+ */
+export async function handler(event, context, callback) {
+	console.log(event);
+
+	try {
+		let user = new User();
+		user.id = event.params.path.id;
+
+		const client = new DyanamoDB();
+		const mapper = new DataMapper({client});
+
+		user = await mapper.get(user).catch(err => {
+			throw (err.name && err.name === 'ItemNotFoundException') ? new ResourceNotFoundException('User', user.id) : err;
```

Comment:
Why not have an Instance of an object rather than passing back a string for an exception?

Reply:
In this case, mapper.get() will throw two types of errors. I feel that the requesting party should only be responsible for providing a valid id, for this endpoint, in order to make changes to the User with that id. If they didn't provide a valid id, then mapper.get() will respond with an error object with the name property set as `ItemNotFoundException`. If that is the case, we want to respond with a 404, and therefore a `ResourceNotFoundException`. However, in any other case, an error thrown is not the responsiblity of the requester, and could have any number of reasons. For that case, we want to pass the error through to be logged to CloudWatch, and returns a generic 500 Error, and we will be alerted via the alarms.

In both cases, an object is used to handle the errors, and I only use string comparison to determine which error is being returned from the library.